File: piwigo/Albums/Headers/CategoryHeaderReusableView.m
//
//  CategoryHeaderReusableView.m
//  piwigo
//
//  Created by Eddy Lelièvre-Berna on 24/06/2018.
//  Copyright © 2018 Piwigo.org. All rights reserved.
//

#import "CategoryHeaderReusableView.h"

@interface CategoryHeaderReusableView()

@end

@implementation CategoryHeaderReusableView

-(instancetype)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if(self)
    {
        self.commentLabel = [UILabel new];
        self.commentLabel.backgroundColor = [UIColor clearColor];
        self.commentLabel.translatesAutoresizingMaskIntoConstraints = NO;
        self.commentLabel.numberOfLines = 0;
        self.commentLabel.adjustsFontSizeToFitWidth = NO;
        self.commentLabel.lineBreakMode = NSLineBreakByWordWrapping;
        self.commentLabel.textAlignment = NSTextAlignmentCenter;
        self.commentLabel.font = [UIFont piwigoFontNormal];
        self.commentLabel.text = @"";

        [self addSubview:self.commentLabel];
        [self addConstraint:[NSLayoutConstraint constraintViewFromTop:self.commentLabel amount:4]];
        if (@available(iOS 11, *)) {
            [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"|-[header]-|"
                                                                           options:kNilOptions
                                                                           metrics:nil
                                                                             views:@{@"header" : self.commentLabel}]];
        } else {
            [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"|-15-[header]-15-|"
                                                                           options:kNilOptions
                                                                           metrics:nil
                                                                             views:@{@"header" : self.commentLabel}]];
        }
    }
    return self;
}

@end

File: piwigo/Album/Headers & Footers/CategoryHeaderReusableView.swift
//
//  CategoryHeaderReusableView.swift
//  piwigo
//
//  Created by Eddy Lelièvre-Berna on 24/06/2018.
//  Copyright © 2018 Piwigo.org. All rights reserved.
//
//  Converted to Swift 5.1 by Eddy Lelièvre-Berna on 14/04/2020
//

import UIKit

class CategoryHeaderReusableView: UICollectionReusableView {
    @objc var commentLabel: UILabel?

    override init(frame: CGRect) {
        super.init(frame: frame)
        self.backgroundColor = UIColor.clear
        
        commentLabel = UILabel()
        commentLabel?.backgroundColor = UIColor.clear
        commentLabel?.translatesAutoresizingMaskIntoConstraints = false
        commentLabel?.numberOfLines = 0
        commentLabel?.adjustsFontSizeToFitWidth = false
        commentLabel?.lineBreakMode = .byWordWrapping
        commentLabel?.textAlignment = .center
        commentLabel?.font = .piwigoFontNormal()
        commentLabel?.text = ""

        if let commentLabel = commentLabel {
            addSubview(commentLabel)
        }
        addConstraint(NSLayoutConstraint.constraintView(fromTop: commentLabel, amount: 4)!)
        if #available(iOS 11, *) {
            if let commentLabel = commentLabel {
                addConstraints(NSLayoutConstraint.constraints(withVisualFormat: "|-[header]-|", options: [], metrics: nil, views: [
                "header": commentLabel
                ]))
            }
        } else {
            if let commentLabel = commentLabel {
                addConstraints(NSLayoutConstraint.constraints(withVisualFormat: "|-15-[header]-15-|", options: [], metrics: nil, views: [
                "header": commentLabel
                ]))
            }
        }
    }

    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
    }
}
